wallet/src/signer/trezor_signer/tests.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::{Add, Div, Mul, Sub};
+
+use super::*;
+use crate::key_chain::{MasterKeyChain, LOOKAHEAD_SIZE};
+use crate::{Account, SendRequest};
+use common::chain::config::create_regtest;
+use common::chain::output_value::OutputValue;
+use common::chain::signature::inputsig::arbitrary_message::produce_message_challenge;
+use common::chain::signature::verify_signature;
+use common::chain::timelock::OutputTimeLock;
+use common::chain::tokens::TokenId;
+use common::chain::{
+    AccountNonce, AccountOutPoint, DelegationId, GenBlock, PoolId, Transaction, TxInput,
+};
+use common::primitives::{Amount, Id, H256};
+use crypto::key::{KeyKind, PrivateKey};
+use randomness::{Rng, RngCore};
+use rstest::rstest;
+use test_utils::random::{make_seedable_rng, Seed};
+use trezor_client::find_devices;
+use wallet_storage::{DefaultBackend, Store, Transactional};
+use wallet_types::account_info::DEFAULT_ACCOUNT_INDEX;
+use wallet_types::seed_phrase::StoreSeedPhrase;
+use wallet_types::KeyPurpose::{Change, ReceiveFunds};
+
+const MNEMONIC: &str =
+    "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about";
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn sign_message(#[case] seed: Seed) {
+    let mut rng = make_seedable_rng(seed);
+
+    let chain_config = Arc::new(create_regtest());
+    let db = Arc::new(Store::new(DefaultBackend::new_in_memory()).unwrap());
+    let mut db_tx = db.transaction_rw_unlocked(None).unwrap();
+
+    let master_key_chain = MasterKeyChain::new_from_mnemonic(
```

Comment:
The current test assumes the device will be initialized with the same mnemonic, but that can be fixed later as the tests can't run automatically now anyway.

To automate the tests I think we will need first to make the signing async, so that we can return from the wallet the status of the signing, i.e. when  it is waiting for a button to be pressed on the trezor device